### DISCUSSION TITLE
Vector.truncate(length)

### DISCUSSION THREAD

**Discussion by feois** [godotengine/godot-proposals#9020] [Engine Core]
Add a built-in method to truncate vector to given length. While it can be easily implemented as `vec * minf(1.0, len / vec.length())`, it's a hassle to write, and if it's built-in, there can be some hacky way to optimize if possible.

**Comment by mieldepoche** [comment]
vectors already have a `limit_length` method, is that what you want?

**Reply by feois** [reply]
... yes, sorry didn't know since the naming is quite different from convention

**Reply by Calinou** [reply]
https://github.com/godotengine/godot/pull/78990 should help with this :slightly_smiling_face:
